storage/storage_image.go

```diff
@@ -64,6 +64,11 @@ type storageImageDestination struct {
 	fileSizes      map[digest.Digest]int64         // Mapping from layer blobsums to their sizes
 	filenames      map[digest.Digest]string        // Mapping from layer blobsums to names of files we used to hold them
 	SignatureSizes []int                           `json:"signature-sizes,omitempty"` // List of sizes of each signature slice
+
+	// data for {Lock,Unlock}Blob
+	lockedDigests   map[digest.Digest]storage.Locker // Bookkeeping of digests that have been locked
+	unlockedDigests map[digest.Digest]bool           // Bookkeeping of digests that have been unlocked. Required to unlock yet unlocked digests in Close()
```

Comment:
(Absolutely non-blocking: The symmetry between `lockedDigests` and `unlockedDigests`, when are so different, makes me nervous. Maybe name the first `digestLockers`?

Also, while the `unlockedDigests` idea does work correctly AFAICT, I can’t quite formulate a _simple_ explanation why. If you have one, writing it down (here or somewhere around `LockBlob`) would be nice.)